src/main/java/net/minestom/server/permission/PermissionHandler.java

```diff
@@ -1,138 +1,17 @@
-     *
-     * @param permissionName the permission name
-     */
-    default void removePermission(@NotNull String permissionName) {
-        getAllPermissions().removeIf(permission -> permission.getPermissionName().equals(permissionName));
-    }
-
     /**
      * Gets if this handler has the permission {@code permission}.
-     * This method will also pattern match for wildcards. For example, if this handler has the permission {@code "*"}, this method will always return true.
-     * However, if this handler has the permission {@code "foo.b*r.baz"}, this method will return true if {@code permission} is {@code "foo.baaar.baz"} or {@code "foo.br.baz}, but not {@code "foo.bar.bz"}.
-     * <p>
-     * Uses {@link Permission#equals(Object)} internally.
      *
      * @param permission the permission to check
      * @return true if the handler has the permission, false otherwise
      */
-    default boolean hasPermission(@NotNull Permission permission) {
-        for (Permission permissionLoop : getAllPermissions()) {
-            if (permissionLoop.equals(permission)) {
-                return true;
-            }
-            String permissionLoopName = permissionLoop.getPermissionName();
-            if (permissionLoopName.contains("*")) {
-                // Sanitize permissionLoopName
-                String regexSanitized = Pattern.quote(permissionLoopName).replace("*", "\\E(.*)\\Q"); // Replace * with regex
-                // pattern matching for wildcards, where foo.b*r.baz matches foo.baaaar.baz or foo.bar.baz
-                if (permission.getPermissionName().matches(regexSanitized)) {
-                    return true;
-                }
-            }
-        }
-        return false;
-    }
-
-    /**
-     * Gets the {@link Permission} with the name {@code permissionName}.
-     * <p>
-     * Useful if you want to retrieve the permission data.
-     *
-     * @param permissionName the permission name
-     * @return the permission from its name, null if not found
-     */
-    @Nullable
-    default Permission getPermission(@NotNull String permissionName) {
-        for (Permission permission : getAllPermissions()) {
-            // Verify permission name equality
-            if (permission.getPermissionName().equals(permissionName)) {
-                return permission;
-            }
-        }
-        return null;
-    }
-
-    /**
-     * Gets if this handler has the permission with the name {@code permissionName} and which verify the optional
-     * {@link PermissionVerifier}.
-     *
-     * @param permissionName     the permission name
-     * @param permissionVerifier the optional verifier,
-     *                           null means that only the permission name will be used
-     * @return true if the handler has the permission, false otherwise
-     */
-    default boolean hasPermission(@NotNull String permissionName, @Nullable PermissionVerifier permissionVerifier) {
```

Comment:
For a none-breaking version, shouldn't this be readded (to this interface) and implemented in DefaultPermissionHandler, so existing code can expect the same results.